### PR TITLE
INC-434: listen for received prisoner and add IEP level for transfer and first arrival

### PR DESCRIPTION
**Pull request by neilmendum**
https://dsdmoj.atlassian.net/browse/INC-434

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonApiService.kt [hunk 1]
```diff
@@ -75,21 +78,37 @@ class PrisonApiService(private val prisonWebClient: WebClient) {
...
       .awaitBodilessEntity()
 
-  suspend fun getPrisonerInfo(prisonerNumber: String): PrisonerAtLocation =
-    prisonWebClient.get()
-      .uri("/api/bookings/offenderNo/$prisonerNumber")
-      .retrieve()
-      .awaitBody()
+  suspend fun getPrisonerInfo(prisonerNumber: String, useClientCredentials: Boolean = false): PrisonerAtLocation {
+    return if (useClientCredentials) {
+      prisonWebClientClientCredentials.get()
```
Comment (Mjwillis) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonApiService.kt:83]: These seem to be the same call, but different web clients, be better to use same code, just different variable

File: src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonApiService.kt [hunk 2]
```diff
@@ -75,21 +78,37 @@ class PrisonApiService(private val prisonWebClient: WebClient) {
...
       .awaitBody()
 
-  suspend fun getLocationById(locationId: Long): Location =
-    prisonWebClient.get()
-      .uri("/api/locations/$locationId")
-      .retrieve()
-      .awaitBody()
+  suspend fun getLocationById(locationId: Long, useClientCredentials: Boolean = false): Location {
+    return if (useClientCredentials) {
+      prisonWebClientClientCredentials.get()
```
Comment (Mjwillis) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonApiService.kt:103]: as above

**Review by Mjwillis — APPROVED** [review]
2 minor issues

File: src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonApiService.kt (hunk 1 above)
Comment (neilmendum) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/incentivesapi/service/PrisonApiService.kt]: Good spot! addressed

**Review by Mjwillis — APPROVED** [review]
LGTM

**Pull request merged**